File: Lista 1/Zadanie 2/WierszTrojkataPascala.cpp
#include <iostream>
#include <string>
#include <exception>
#include <sstream>

class BlednyArgumentKonstruktora : public std::exception
{
    private:
    std::string blad;

    public:
    BlednyArgumentKonstruktora(int n)
    {
       blad = "Argument: " + std::to_string(n) + " nie jest poprawnym argumentem. Argument powinien byc liczba dodatnia.";
    }

    virtual const char* what() const noexcept
    {
        return blad.c_str();
    }
};

class BlednyArgumentMetody : public std::exception
{
    private:
    std::string blad;

    public:
    BlednyArgumentMetody(int n, int m)
    {
        blad = "Argument: " + std::to_string(m) + " nie jest poprawnym argumentem metody. Argument powinien byc w zakresie [0, " + std::to_string(n) + "]";
    }

    virtual const char* what() const noexcept
    {
        return blad.c_str();
    }
};


class WierszTrojkataPascala
{
    private:
    int *wiersz = nullptr;
    int n;

    public:
    WierszTrojkataPascala(int n) throw(BlednyArgumentKonstruktora)
    {
        if(n < 0)
            throw BlednyArgumentKonstruktora(n);
        
        wiersz = new int[n+1];
        this->n = n;

        for(int i = 0; i < n+1; i++) 
        {
            wiersz[i] = 1;
            for(int j = i - 1; j > 0; j--) 
            {
                wiersz[j] += wiersz[j-1];
            } 
        }
    }

    ~WierszTrojkataPascala()
    {
        if(wiersz != nullptr)
            delete[] wiersz;
    }

    int wspolczynnik(int m) throw(BlednyArgumentMetody)
    {
        if(m < 0 || m > n)
            throw BlednyArgumentMetody(n, m);
        return wiersz[m];
    }
};



int main(int argc, char* argv[])
{
    if(argc < 2)
    {
        std::cerr << "Niepoprawna ilosc argumentow!\n";
        return 1;
    }
    
    std::istringstream ss;
	ss.str(argv[1]);
    int n;
    ss >> n;

    if(ss.fail() || !ss.eof())
    {
        std::cerr << "Argument nie jest liczba naturalna!\n";
        return 2;
    }

    try 
    {
        WierszTrojkataPascala wierszTrojkata(n);
        int m;

        for(int i = 2; i < argc; i++)
        {
            ss.clear();
            ss.str(argv[i]);
            ss >> m;
            if(ss.fail() || !ss.eof())
            {
                std::cerr << "Argument nie jest liczba naturalna!\n";
                continue;
            }

            try
            {
                std::cout << wierszTrojkata.wspolczynnik(m) << std::endl;
            }
            catch(BlednyArgumentMetody ex)
            {
                std::cerr << ex.what() << std::endl;
            }

        }        
    }
    catch(BlednyArgumentKonstruktora ex) 
    {
        std::cerr << ex.what() << std::endl;
    }
    catch(std::exception ex)
    {
        std::cerr << ex.what();
    }
    catch(...)
    {}

    return 0;
}

File: Lista 1/Zadanie 1/WierszTrojkataPascala.java
public class WierszTrojkataPascala {

    private int[] wiersz;
    private int n;
    public WierszTrojkataPascala(int n) throws BlednyArgumentKonstruktora {
        if(n < 0)
            throw new BlednyArgumentKonstruktora(n);
        
        this.n = n;
        wiersz = new int[n + 1];

        for(int i = 0; i < wiersz.length; i++) {
            wiersz[i] = 1;
            for(int j = i - 1; j > 0; j--) {
                wiersz[j] += wiersz[j-1];
            } 
        }
    }

    public int wspolczynnik(int m) throws BlednyArgumentMetody {
        if(m < 0 || m > n)
            throw new BlednyArgumentMetody(n, m);
        return wiersz[m];
    }
}

class BlednyArgumentKonstruktora extends Exception {
    public BlednyArgumentKonstruktora(int n) {
        super("Argument: " + n + " nie jest poprawnym argumentem. Argument powinien byc liczba dodatnia.");
        //Zmienic po policzeniu
    }
}

class BlednyArgumentMetody extends Exception {
    public BlednyArgumentMetody(int n, int m) {
        super("Argument: " + m + " nie jest poprawnym argumentem metody. Argument powinien byc w zakresie [0, " + n + "]");
    }
}